[GTK][WPE] Remove support for named spaces and nonzero unitless lengths

fred-wang opened the pull request:
#### f243e31b7e947af375a58008b99319e2b0175dff
<pre>
[GTK][WPE] Remove support for named spaces and nonzero unitless lengths
<a href="https://bugs.webkit.org/show_bug.cgi?id=197494">https://bugs.webkit.org/show_bug.cgi?id=197494</a>

Reviewed by Rob Buis.

This commit removes support for named spaces and nonzero unitless
lengths in WebKit when the MathML Core flag is enabled [1] [2],
which is currently the case on GTK and WPE platforms. It also adds
a few more WPT tests to cover attributes implemented in WebKit.

[1] <a href="https://developer.mozilla.org/en-US/docs/Web/MathML/Attribute/Values#legacy_mathml_lengths">https://developer.mozilla.org/en-US/docs/Web/MathML/Attribute/Values#legacy_mathml_lengths</a>
[2] <a href="https://developer.mozilla.org/en-US/docs/Web/MathML/Element/mpadded#legacy_syntax">https://developer.mozilla.org/en-US/docs/Web/MathML/Element/mpadded#legacy_syntax</a>

* LayoutTests/TestExpectations: Remove a test fixed by this change.
* LayoutTests/accessibility/mac/mathml-line-fraction.html: This test uses a nonzero unitless value, so disable the MathML Core.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/fractions/frac-linethickness-001.html: enable MathML Core.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/fractions/frac-linethickness-002-expected.txt: Update expectation.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/fractions/frac-linethickness-002.html: enable MathML Core.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/mpadded/mpadded-legacy-length-values-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/mpadded/mpadded-legacy-length-values.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/mo-lspace-rspace-legacy-length-values-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/mo-lspace-rspace-legacy-length-values.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/mo-minsize-maxsize-legacy-length-values-expected.txt: Added.
  There seem to be an existing stretching bug unrelated to this change that make maxsize tests fail.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/mo-minsize-maxsize-legacy-length-values.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/spaces/mspace-legacy-length-values-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/spaces/mspace-legacy-length-values.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/lengths-2-expected.txt: Update expectation.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/css-styling/lengths-2-expected.txt: Ditto.
* LayoutTests/mathml/non-core/positive-namedspace.html: Renamed from LayoutTests/mathml/presentation/positive-namedspace.html.
  This test is specifically tested namedspace so disable disable the MathML Core. Also move it to non-core directory.
* LayoutTests/mathml/non-core/positive-namedspace-expected.html: Ditto.
* LayoutTests/mathml/presentation/mo-form-expected.html: This test uses namedspaces but actually does not need to do so, so tweak it.
* LayoutTests/mathml/presentation/non-bmp-operators-spacing-expected.html: Ditto.
* LayoutTests/mathml/presentation/non-bmp-operators-stretching-expected.html: Ditto.
* Source/WebCore/mathml/MathMLFractionElement.cpp:
(WebCore::MathMLFractionElement::lineThickness): Pass boolean to indicate whether to support legacy MathML lengths.
* Source/WebCore/mathml/MathMLPresentationElement.cpp:
(WebCore::MathMLPresentationElement::parseNumberAndUnit): Make parsing fails for nonzero unitless.
(WebCore::MathMLPresentationElement::parseMathMLLength): Make parsing fails for namedspace and pass boolean to indicate whether to support
  legacy MathML lengths.
(WebCore::MathMLPresentationElement::cachedMathMLLength):
* Source/WebCore/mathml/MathMLPresentationElement.h: Add new boolean to indicate whether to support legacy MathML lengths.

Canonical link: <a href="https://commits.webkit.org/255818@main">https://commits.webkit.org/255818@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c02b45bc60aaac324722041b6f8f94963728a92b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/93706 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/2901 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/24317 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/103350 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/163670 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/97699 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/2914 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/31159 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/86045 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/99394 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/99367 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/2072 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/80144 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/29096 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/83996 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/24317 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/72048 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/37550 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/24317 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/35400 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/24317 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4021 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/39277 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/41369 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/41212 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/24317 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
<!--EWS-Status-Bubble-End-->

rwlbuis commented on Source/WebCore/mathml/MathMLPresentationElement.h:
```diff
@@ -60,7 +60,7 @@ class MathMLPresentationElement : public MathMLElement {
     bool isPresentationMathML() const final { return true; }
 
-    static Length parseNumberAndUnit(StringView);
+    static Length parseNumberAndUnit(StringView, bool coreMathMLEnabled);
```
Is coreMathMLEnabled accurate? Is it not more a question of support mode for unitless/named spaces? I think the name could be better.

fred-wang replied:
Right, I guess I can rename.

rwlbuis reviewed (approved):
LGTM with one nit.

fred-wang committed:
[GTK][WPE] Remove support for named spaces and nonzero unitless lengths
https://bugs.webkit.org/show_bug.cgi?id=197494

Reviewed by Rob Buis.

This commit removes support for named spaces and nonzero unitless
lengths in WebKit when the MathML Core flag is enabled [1] [2],
which is currently the case on GTK and WPE platforms. It also adds
a few more WPT tests to cover attributes implemented in WebKit.

[1] https://developer.mozilla.org/en-US/docs/Web/MathML/Attribute/Values#legacy_mathml_lengths
[2] https://developer.mozilla.org/en-US/docs/Web/MathML/Element/mpadded#legacy_syntax

* LayoutTests/TestExpectations: Remove a test fixed by this change.
* LayoutTests/accessibility/mac/mathml-line-fraction.html: This test uses a nonzero unitless value, so disable the MathML Core.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/fractions/frac-linethickness-001.html: enable MathML Core.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/fractions/frac-linethickness-002-expected.txt: Update expectation.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/fractions/frac-linethickness-002.html: enable MathML Core.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/mpadded/mpadded-legacy-length-values-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/mpadded/mpadded-legacy-length-values.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/mo-lspace-rspace-legacy-length-values-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/mo-lspace-rspace-legacy-length-values.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/mo-minsize-maxsize-legacy-length-values-expected.txt: Added.
  There seem to be an existing stretching bug unrelated to this change that make maxsize tests fail.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/operators/mo-minsize-maxsize-legacy-length-values.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/spaces/mspace-legacy-length-values-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/mathml/presentation-markup/spaces/mspace-legacy-length-values.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/lengths-2-expected.txt: Update expectation.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/css-styling/lengths-2-expected.txt: Ditto.
* LayoutTests/mathml/non-core/positive-namedspace.html: Renamed from LayoutTests/mathml/presentation/positive-namedspace.html.
  This test is specifically tested namedspace so disable disable the MathML Core. Also move it to non-core directory.
* LayoutTests/mathml/non-core/positive-namedspace-expected.html: Ditto.
* LayoutTests/mathml/presentation/mo-form-expected.html: This test uses namedspaces but actually does not need to do so, so tweak it.
* LayoutTests/mathml/presentation/non-bmp-operators-spacing-expected.html: Ditto.
* LayoutTests/mathml/presentation/non-bmp-operators-stretching-expected.html: Ditto.
* Source/WebCore/mathml/MathMLFractionElement.cpp:
(WebCore::MathMLFractionElement::lineThickness): Pass boolean to indicate whether to support legacy MathML lengths.
* Source/WebCore/mathml/MathMLPresentationElement.cpp:
(WebCore::MathMLPresentationElement::parseNumberAndUnit): Make parsing fails for nonzero unitless.
(WebCore::MathMLPresentationElement::parseMathMLLength): Make parsing fails for namedspace and pass boolean to indicate whether to support
  legacy MathML lengths.
(WebCore::MathMLPresentationElement::cachedMathMLLength):
* Source/WebCore/mathml/MathMLPresentationElement.h: Add new boolean to indicate whether to support legacy MathML lengths.

Canonical link: https://commits.webkit.org/255818@main
